tests/test_types.py

```diff
@@ -3927,23 +3927,25 @@ class Foobar(BaseModel):
         empty_password: SecretStr
 
     # Initialize the model.
-    f = Foobar(password='1234', empty_password='')
-
-    # Assert correct types.
-    assert f.password.__class__.__name__ == 'SecretStr'
-    assert f.empty_password.__class__.__name__ == 'SecretStr'
-
-    # Assert str and repr are correct.
-    assert str(f.password) == '**********'
-    assert str(f.empty_password) == ''
-    assert repr(f.password) == "SecretStr('**********')"
-    assert repr(f.empty_password) == "SecretStr('')"
-    assert len(f.password) == 4
-    assert len(f.empty_password) == 0
-
-    # Assert retrieval of secret value is correct
-    assert f.password.get_secret_value() == '1234'
-    assert f.empty_password.get_secret_value() == ''
+    f1 = Foobar(password='1234', empty_password='')
+    f2 = Foobar.model_validate_json('{"password": "1234", "empty_password": ""}')
+
+    for f in [f1, f2]:
```

Comment:
It may be slightly nicer to turn this into a parameterized test rather than have one of `f1` or `f2` fail inside the loop